Std/Data/String/Basic.lean

```diff
@@ -1,15 +1,63 @@
 /-
 Copyright (c) 2022 Jannis Limperg. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Jannis Limperg
+Authors: Jannis Limperg, James Gallicchio, F. G. Dorais
 -/
 
 import Std.Data.Char
 import Std.Data.Nat.Lemmas
+import Std.Data.Array.Match
 
-protected theorem String.Pos.ne_zero_of_lt : {a b : Pos} → a < b → b ≠ 0
+namespace String
+
+instance : Coe String Substring := ⟨toSubstring⟩
```

Comment:
This should be a separate PR, which also removes most uses of `toSubstring` in the rest of the code at the same time

Reply:
#219